graylog2-server/src/main/java/org/graylog2/indexer/searches/Searches.java

```diff
@@ -252,66 +264,72 @@ public SearchResult search(String query, String filter, TimeRange range, int lim
     }
 
     public SearchResult search(SearchesConfig config) {
-        Set<IndexRange> indices = determineAffectedIndicesWithRanges(config.range(), config.filter());
+        final Set<IndexRange> indexRanges = determineAffectedIndicesWithRanges(config.range(), config.filter());
+        final Set<String> indices = indexRanges.stream().map(IndexRange::indexName).collect(Collectors.toSet());
 
-        Set<String> indexNames = Sets.newHashSet();
-        for (IndexRange index : indices) {
-            indexNames.add(index.indexName());
-        }
+        final SearchSourceBuilder requestBuilder = searchRequest(config);
+
+        final Search.Builder searchBuilder = new Search.Builder(requestBuilder.toString())
+            .addType(IndexMapping.TYPE_MESSAGE)
+            .addIndex(indices);
 
-        SearchRequest request = searchRequest(config, indexNames).request();
+        final io.searchbox.core.SearchResult searchResult = execute(searchBuilder.build());
+        final List<ResultMessage> hits = searchResult.getHits(Map.class).stream()
+            .map(hit -> ResultMessage.parseFromSource(hit.id, hit.index, (Map<String, Object>)hit.source))
+            .collect(Collectors.toList());
+        final long tookMs = tookMsFromSearchResult(searchResult);
+        recordEsMetrics(tookMs, config.range());
 
-        SearchResponse r = c.search(request).actionGet();
-        recordEsMetrics(r, config.range());
+        return new SearchResult(hits, indexRanges, config.query(), requestBuilder.buildAsBytes(), new TimeValue(tookMs));
+    }
 
-        return new SearchResult(r.getHits(), indices, config.query(), request.source(), r.getTook());
+    private long tookMsFromSearchResult(io.searchbox.core.SearchResult searchResult) {
+        return new Double(searchResult.getValue("took").toString()).longValue();
```

Comment:
This will fail with a `NullPointerException` if `took` doesn't exist.
The question is, if the `took` field always exists. 🤔